src/providers/s3Provider.ts

```diff
@@ -0,0 +1,71 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import { DeleteObjectCommand, HeadObjectCommand, S3Client } from '@aws-sdk/client-s3';
+import { Logger } from '@map-colonies/js-logger';
+import httpStatus from 'http-status-codes';
+import { inject, injectable } from 'tsyringe';
+import { Provider, S3Config } from '../common/interfaces';
+import { AppError } from '../common/appError';
+import { SERVICES } from '../common/constants';
+
+@injectable()
+export class S3Provider implements Provider {
+  private readonly s3: S3Client;
+
+  public constructor(
+    @inject(SERVICES.PROVIDER_CONFIG) protected readonly s3Config: S3Config,
+    @inject(SERVICES.LOGGER) protected readonly logger: Logger
+  ) {
+    this.s3 = this.createS3Instance(s3Config);
+  }
+
+  public async deleteFile(filePath: string): Promise<void> {
+    const bucketName = this.s3Config.bucket;
+    const result = await this.isFileExist(filePath);
```

Comment:
Don't name the constant as result. It is not best practice. Use more informative name.

Reply:
function and variable removed